### PR TITLE
RavenDB-16064  Fix admin logs filtering

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts [hunk 1]
```diff
@@ -183,31 +183,33 @@ class adminLogs extends viewModelBase {
...
-            
-            const filteredItems = fromFilterChange
-                ? this.allData.filter(filterFunction)
-                : this.pendingMessages.filter(filterFunction);
-            
-            this.listController().pushElements(filteredItems);
-        } else {
-            this.listController().pushElements(fromFilterChange ? this.allData : this.pendingMessages);
         }
+        return x => true;
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: return null

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts [hunk 2]
```diff
@@ -183,31 +183,33 @@ class adminLogs extends viewModelBase {
...
-        const errorsOnly = this.onlyErrors();
-
         if (fromFilterChange) {
             this.listController().reset();
         }
 
+        const filterFunction = this.getFilterFunction();
+
+        const filteredItems = fromFilterChange ?
+            this.allData.filter(filterFunction) :
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: filter only if filter function is not null

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts [hunk 3]
```diff
@@ -348,16 +350,19 @@ class adminLogs extends viewModelBase {
...
+            this.listController().pushElements([...filteredItems, msg]);
+            this.pendingMessages.length = 0;
+            
         } else {
             this.pendingMessages.push(msg);
         }
     }
     
-    private onAppendPendingMessages() {
+    private onMessageAdded() {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: name is wrong ihmo.

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: done

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: done

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: we used to just append to `this.pendingMessages`
but now there is also the `showMessageNow `flow
and then we don't append to `this.pendingMesages`, but **add** to `listController`

**Review by ml054 — APPROVED** [review]

**Pull request merged**